improver/utilities/temporal_interpolation.py

```diff
@@ -388,6 +553,36 @@ def process(self, cube_t0: Cube, cube_t1: Cube) -> CubeList:
             )
             raise TypeError(msg)
 
+        if cube_t0.coord("time").bounds is not None:
+            if self.interpolation_method not in self.period_interpolation_methods:
+                raise ValueError(
+                    "Period diagnostics can only be temporally interpolated "
+                    f"using these methods: {self.period_interpolation_methods}.\n"
+                    f"Currently selected method is: {self.interpolation_method}."
+                )
+            if not any([self.accumulation, self.max, self.min]):
+                raise ValueError(
+                    "A type of period must be specified when interpolating a "
+                    "period diagnostic. This may be a period max, min"
+                    " or an accumulation."
+                )
+            cube_interval = (
+                cube_t1.coord("time").points[0] - cube_t0.coord("time").points[0]
+            )
+            (period,) = np.diff(cube_t1.coord("time").bounds[0])
+            if not cube_interval == period:
+                raise ValueError(
+                    "The diagnostic provided represents the period "
+                    f"{period / 3600} hours. The interval between the "
+                    f"diagnostics is {cube_interval / 3600} hours. Temporal "
+                    "interpolation can only be applied to a period "
+                    "diagnostic provided at intervals that match the "
+                    "diagnostic period such that all points in time are "
+                    "captured by only one of the inputs and do not overlap."
+                )
+
+            self.period_inputs = True
```

Comment:
I agree. This grew organically in some regards and the structure is not ideal. I'll rejig more of this into the `__init__` method and rejig the tests.

Reply:
The modifications are such that:
- providing one of the min, max, accumulation flags sets the `period_inputs` variable, rather than the cube having bounds.
- the interpolation method is now tested in the `__init__` method.
- the test for failing to declare a type of period when bounds are found has been removed as the presence of the type is now indicative of working with a period.
- a new test has been added to process to check that the inputs do have time bounds if a type of period has been declared (which implies period_inputs = True).